src/lib.rs

```diff
@@ -0,0 +1,436 @@
+                                    permissions: Option<JsonValue>)
+                                    -> Result<JsonValue, KintoError> {
+
+        let endpoint = format!("{}/buckets/{bucket_id}/groups/{group_id}",
+                               self.server_url, bucket_id=bucket_id, group_id=group_id);
+        let payload = self.set_payload(data, permissions);
+        return self.handle_request(Method::Post, endpoint, payload.into());
+    }
+
+    pub fn patch_group(&self, bucket_id: String, group_id: String, data: Option<JsonValue>)
+                        -> Result<JsonValue, KintoError> {
+        return self.patch_group_with_perms(bucket_id, group_id, data, None);
+    }
+
+    pub fn patch_group_with_perms(&self, bucket_id: String, group_id: String,
+                                   data: Option<JsonValue>,
+                                   permissions: Option<JsonValue>)
+                                   -> Result<JsonValue, KintoError> {
+
+        let endpoint = format!("{}/buckets/{bucket_id}/groups/{group_id}",
+                               self.server_url, bucket_id=bucket_id, group_id=group_id);
+        let payload = self.set_payload(data, permissions);
+        return self.handle_request(Method::Post, endpoint, payload.into());
+    }
+
+
+    pub fn set_payload(&self, data: Option<JsonValue>,
+                       permissions: Option<JsonValue>) -> JsonValue {
+
+        let mut payload = JsonValue::new_object();
+
+        payload["data"] = data.unwrap_or(JsonValue::new_object());
+        payload["permissions"] = permissions.unwrap_or(JsonValue::new_object());
+
+        return payload;
+    }
+
+    pub fn handle_request(&self, method: Method, endpoint: String,
+                          payload: Option<JsonValue>)
+                          -> Result<JsonValue, KintoError> {
+
+        let request = self.http_client
+            .request(method, &endpoint)
+            .headers(self.headers.to_owned())
+            .body(payload.unwrap_or(JsonValue::new_object()).dump().as_str())
+            .send();
+
+        let mut response = match request {
+            Ok(response) => response,
+            Err(_) => return Err(KintoError::HyperError),
+        };
+
+        let mut body = String::new();
+        response.read_to_string(&mut body).unwrap();
+        let data = json::parse(&body);
+
+        match data {
+            Ok(response) => return Ok(response),
+            Err(_) => return Err(KintoError::JsonError),
+        };
+    }
+
+   // Collections
+
+    pub fn get_collections(&self, bucket_id: String)
+                      -> Result<JsonValue, KintoError> {
+        let endpoint = format!("{}/buckets/{bucket_id}/collections",
+                               self.server_url, bucket_id=bucket_id);
+        return self.handle_request(Method::Get, endpoint, None);
+    }
+
+    pub fn delete_collections(&self, bucket_id: String)
+                         -> Result<JsonValue, KintoError> {
```

Comment:
What does a DELETE request normally return? Is there any real useful information in it? Maybe we should just have it return `Result<(), KintoError>`.